### PR TITLE
update python versions to allow v4.4.0-devel format

### PR DESCRIPTION
**Pull request by jsonar-cpapke**
Also removing checks for the version of setuptools, as version `8.0` only supported python up to version 3.4 and so can't be used by us anyway.

### CODE REVIEW

**Review by yhager — APPROVED** [review]

File: SonarFunctions.cmake [hunk 1]
```diff
@@ -71,27 +71,20 @@ endfunction()
...
 function(sonar_python_version python_version_var)
   cmake_parse_arguments(PARSE_ARGV 1 PYTHON "" "SETUPTOOLS_VERSION" "")
-  if(NOT PYTHON_SETUPTOOLS_VERSION)
-    set(PYTHON_SETUPTOOLS_VERSION "0.9.8")
-  endif()
   sonar_set_version(version)
-  set(regex "([0-9]+\.[0-9]+\.[0-9]+)-([0-9]+)-(.*)")
-  if(version MATCHES ${regex})
+  set(devel_regex "([0-9]+\\.[0-9]+(\\.[0-9]+)*)-devel(op)?(-.*)?")
+  set(regex "([0-9]+\\.[0-9]+(\\.[0-9]+)*)(-.*)?")
```
Comment (jsonar-cpapke) [inline SonarFunctions.cmake:76]: So.... turns out you need a double back slash for the regex to work correctly, and so I don't think this ever worked.....

**Comment by jsonar-cpapke** [comment]
I did check that this works and it seems to tag things correctly.

**Review by oorijsonar — APPROVED** [review]

File: SonarFunctions.cmake (hunk 1 above)
Comment (jsonar-cpapke) [reply SonarFunctions.cmake:76]: And to counter myself, that makes no sense, but this seems to work so 🤷

Comment (yhager) [reply SonarFunctions.cmake:76]: I think you're right. Not sure how this ever worked

File: SonarFunctions.cmake [hunk 2]
```diff
@@ -71,27 +71,20 @@ endfunction()
 
 function(sonar_python_version python_version_var)
   cmake_parse_arguments(PARSE_ARGV 1 PYTHON "" "SETUPTOOLS_VERSION" "")
```
Comment (yhager) [inline SonarFunctions.cmake:73]: do we still need the `SETUPTOOLS_VERSION` argument?

File: SonarFunctions.cmake [hunk 3]
```diff
@@ -13,7 +13,7 @@ function(sonar_cpack_version major_var minor_var patch_var)
   string(REGEX REPLACE "${regex}" "\\2" minor "${version}")
   string(REGEX REPLACE "${regex}" "\\3" patch "${version}")
   if (NOT SHORT_VERSION)
-    string(REGEX REPLACE "${regex}" "\\4" extra "${version}")
+    string(REGEX REPLACE "${regex}" "\\5" extra "${version}")
```
Comment (yhager) [inline SonarFunctions.cmake:16]: If this matches the `pre-release` section as per semver, let's call it that

File: SonarFunctions.cmake [hunk 4]
```diff
@@ -2,7 +2,7 @@ include(GetGitRevisionDescription)
 
 function(sonar_cpack_version major_var minor_var patch_var)
   git_describe(version --match "v*")
-  set(regex "v([0-9]+)\.([0-9]+)\.([0-9]+)(.*)")
+  set(regex "v([0-9]+)\\.([0-9]+)\\.([0-9]+(\\.[0-9]+)*)(.*)")
```
Comment (yhager) [inline SonarFunctions.cmake:5]: I think the fourth section should match semver pre-release section. see https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions

File: SonarFunctions.cmake (hunk 2 above)
Comment (jsonar-cpapke) [reply SonarFunctions.cmake:73]: That looks like a no. I don't see it used anywhere else in the cmake project.

File: SonarFunctions.cmake (hunk 3 above)
Comment (jsonar-cpapke) [reply SonarFunctions.cmake]: Done

File: SonarFunctions.cmake (hunk 4 above)
Comment (jsonar-cpapke) [reply SonarFunctions.cmake]: Unless you really want that, the regex we currently have is much easier to understand and will match all the required options.

Comment (yhager) [reply SonarFunctions.cmake]: I don't really want that, no. But I want us to allow semver, and I don't want to fix bugs that come from invalid semver tags

**Comment by jsonar-cpapke** [comment]
@yhager Attempt 3, please have a look

File: SonarFunctions.cmake [hunk 5]
```diff
@@ -2,7 +2,7 @@ include(GetGitRevisionDescription)
 
 function(sonar_cpack_version major_var minor_var patch_var)
   git_describe(version --match "v*")
-  set(regex "v([0-9]+)\.([0-9]+)\.([0-9]+)(.*)")
+  set(regex "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)(-.*|)$")
```
Comment (yhager) [inline SonarFunctions.cmake:5]: Can this work? I am trying to remove the dash from the pre-release group
```suggestion
  set(regex "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)(-(.*))?$")
```

Comment (jsonar-cpapke) [reply SonarFunctions.cmake:5]: The reason I didn't want to do that was I wasn't sure how to conditionally add a `-` into the final version based on if that capture group matches or not. If you know how that works, then that should be fine.

Comment (yhager) [reply SonarFunctions.cmake:5]: I see. Okay, go with your version then. Let's wrap this up.

Comment (jsonar-cpapke) [reply SonarFunctions.cmake:5]: Does that mean merge this?

Comment (yhager) [reply SonarFunctions.cmake:5]: yeah :)

**Pull request merged**